script/ensure-node-source-fixture-files.js

```diff
@@ -0,0 +1,164 @@
+
+const fs = require("fs");
+const path = require("path");
+const https = require("https");
+
+// This "nodeVersion" specifies the version of Node used by certain specs.
+// Update nodeVersion here to run those specs against newer Node source files.
+// (May be needed to work around build issues with newer Node bundled with ppm.)
+// This string is also hard-coded in those spec files, so update it there, too.
+const nodeVersion = "v18.12.1";
+
+const filesToFetch = [
+  {
+    url: `https://nodejs.org/dist/${nodeVersion}/node-${nodeVersion}.tar.gz`,
+    filename: `node-${nodeVersion}.tar.gz`,
+    sha256sum: "ba8174dda00d5b90943f37c6a180a1d37c861d91e04a4cb38dc1c0c74981c186"
+  },
+  {
+    url: `https://nodejs.org/dist/${nodeVersion}/node-${nodeVersion}-headers.tar.gz`,
+    filename: `node-${nodeVersion}-headers.tar.gz`,
+    sha256sum: "9d55ee072ba6d5a141db092cef1a0f715f7d3fc938285a6d927a1d0a0c7442f7"
+  },
+  {
+    url: `https://nodejs.org/dist/${nodeVersion}/win-x64/node.lib`,
+    filename: "node_x64.lib",
+    sha256sum: "1bd376a23d181d85096d1a9c46e6be7fcd20d30f9b8f77a2a847d3dbff8e25c7"
+  },
+  {
+    url: `https://nodejs.org/dist/${nodeVersion}/win-x86/node.lib`,
+    filename: "node.lib",
+    sha256sum: "b1c6dc670911d85ef1704fa56f4cc4c7e1071f4869778398e6d88b3b0b565978"
+  },
+  {
+    url: `https://nodejs.org/dist/${nodeVersion}/SHASUMS256.txt`,
+    filename: "SHASUMS256.txt",
+    sha256sum: "64aad1211a6003dd6529ebf9f19167769d0356ce5affc4245bb26c35aa66a9ed"
+  },
+] // If you ever need to update to a newer Node version, just calculate the sha256sums
+  // for the new files and update them here to match the new files.
+
+const sourceFixtureDir = path.resolve(__dirname, "..", "spec", "fixtures", "node-source");
+fs.mkdirSync(sourceFixtureDir, { recursive: true });
+
+for (const details of filesToFetch) {
+  ensureFile(details);
+} // This is the main loop, with support functions below.
+
+async function ensureFile (details) {
+  // We check if the file already exists, and if so,
+  // make sure its sha256sum matches the expected/correct value.
+  // If it doesn't exist or has the wrong hash, try to re-download.
+
+  logVerbose("details is:");
+  logVerbose(details);
+
+  const destinationPath = path.resolve(sourceFixtureDir, details.filename);
+  logVerbose(`destinationPath is: `${destinationPath}`);
```

Comment:
```suggestion
  logVerbose(`destinationPath is: ${destinationPath}`);
```

Looks like this is causing the rest of the file to be a template literally for some time, and also causing test failures